File: Algorithmen/Fibonacci/cpp/Fibonacci.cpp
#include <iostream>
#include <ctime>

using namespace std;

int main()
{

    int n, m, iterations, i;
    int_fast64_t fib;
    int *ptr_n = &n, *ptr_m = &m;

    cout << "Enter the first number: ";
    cin >> n;
    cout << "Enter the second number: ";
    cin >> m;
    cout << "Enter the number of iterations: ";
    cin >> iterations;

    double start = clock();

    //calculation

    for (i = 0; i < iterations; i++)
    {
        fib = *ptr_n + *ptr_m;
        *ptr_n = *ptr_m;
        *ptr_m = fib;

        if (fib < 0)
        {
            cout << "Integer Overflow";
            break;
        }
        else
        {
            cout << i << "\t" <<  fib << endl;
        }

    }

    double end = clock();

    cout << "\nTime: " << (end - start) / 1000 * CLOCKS_PER_SEC << "ms" << endl;

}

File: Algorithmen/Fibonacci/fortran/Fibonacci.f90
program fibonacci
    implicit none
    
    !boilerplate
    integer :: n, m, iterations, i
    integer(kind=8) :: fib

    !input
    write(*,*) "Enter the first number"
    read(*,*) n

    write(*,*) "Enter the second number"
    read(*,*) m

    write(*,*) "Enter the number of iterations"
    read(*,*) iterations

    !calculation

    i = 0
    do while (i < iterations)
        fib = n + m
        n = m
        m = fib
        i = i + 1

        if (fib < 0) then
            write(*,*) "Integer Overflow"
            exit
        end if

        write(*,*) i, fib
    end do

end program fibonacci